MaintenanceManager/MaintenanceManager.cpp

```diff
@@ -356,6 +394,127 @@ namespace WPEFramework {
+
+		    thunder_client=getThunderPluginHandle(secMgr_callsign_ver);
+                    if (thunder_client == nullptr) {
+                        LOGERR("Failed to get plugin handle\n");
+                    } else {
+                        JsonObject params;
+                        JsonObject joGetResult;
+
+                        params["clientId"] = "gdi";
+                        params["distributedTraceType"] = "money";
+                        params["distributedTraceId"] = "trace-id=8e44ec3c-7aa1-404d-a1e5-1dde72114ca3;parent-id=123;span-id=456";
+
+                        thunder_client->Invoke<JsonObject, JsonObject>(5000, "getDeviceInitializationContext", params, joGetResult);
+			if (joGetResult.HasLabel("success") && joGetResult["success"].Boolean()) 
+                        {
+                                if (joGetResult.HasLabel("partnerProvisioningContext")) {
+                                    JsonObject getProvisioningContext = joGetResult["partnerProvisioningContext"].Object();
+					
+                                    int size = (int)(sizeof(deviceInitializationContext)/sizeof(deviceInitializationContext[0]));
+                                    for (int i=0; i < size; i++) {
+                                        const char* key = deviceInitializationContext[i].c_str();
+
+                                        // Retrive partnerProvisioningContext Value
+                                        string paramValue=getProvisioningContext[key].String();
+					    
+                                        if(!paramValue.empty()) {
+                                            if (strcmp(key, "regionalConfigService") == 0) {
+                                                paramValue = "https://" + paramValue;
+                                            }
+                                            LOGINFO("%s : %s", key, paramValue.c_str());
+					    
+                                            // Retrieve tr181 parameter from m_param_map
+                                            string rfc_parameter = m_param_map[key];
+
+                                            //  Retrieve parameter data type from m_paramType_map
+                                            DATA_TYPE rfc_dataType = m_paramType_map[key];
+
+                                            // Set the RFC values for partnerProvisioningContext parameters
+                                            setRFC(rfc_parameter.c_str(), paramValue.c_str(), rfc_dataType);
+                                        } else {
+                                            LOGINFO("Not able to fetch %s value from partnerProvisioningContext", key);
+                                        }
+                                    }
+                                    success = true;
+				} else {
+                                        LOGINFO("partnerProvisioningContext is not available from response. Retrying in %d seconds", retryDelay);
```

Comment:
partnerProvisioningContext is not available in the response. Retrying in %d seconds